### PR TITLE
fix: check service name contains allowed characters and errors cleanly

### LINKED ISSUE #164: [bug]: A service rejected for not matching the name regex isn't clear what's actually going wrong
### What version of the Kurtosis CLI were you running when you encountered this bug?

0.68.7

### (Optional) What is your metrics user ID?

0497e811755b8f775ef0d99cd22b1ef221708b5a95119857a310cbc4f2eabcae

### What is the severity of this bug?

Papercut, this bug is frustrating, but I have a workaround.

### Description & steps to reproduce

I'm running the following SL:
```
    postgres = plan.add_service(
        "postgres:15.2-alpine",
        ServiceConfig(
            image = "postgres",
            ports = {
                POSTGRES_PORT_ID: PortSpec(5432, application_protocol = "postgresql"),
            },
            env_vars = {
                "POSTGRES_PASSWORD": POSTGRES_PASSWORD,
            },
        ),
    )
```
As you can see, I accidentally swapped the image & service names. This threw an error because `:` isn't part of the valid regex.

### Relevant log output

```true
WARN[2023-03-14T12:28:58-03:00] You are running an old version of the Kurtosis CLI; we suggest you to update it to the latest version, '0.68.8'
WARN[2023-03-14T12:28:58-03:00] You can manually upgrade the CLI tool following these instructions: https://docs.kurtosis.com/install#upgrading
INFO[2023-03-14T12:28:58-03:00] Creating a new enclave for Starlark to run inside...
INFO[2023-03-14T12:29:01-03:00] Enclave 'whispering-ravine' created successfully

> add_service service_name="postgres:15.2-alpine" config=ServiceConfig(image="postgres", ports={"postgres": PortSpec(number=5432, application_protocol="postgresql")}, env_vars={"POSTGRES_PASSWORD": "password"})
There was an error executing Starlark code
An error occurred executing instruction (number 1) at DEFAULT_PACKAGE_ID_FOR_SCRIPT[5:32]:
add_service(service_name="postgres:15.2-alpine", config=ServiceConfig(image="postgres", ports={"postgres": PortSpec(number=5432, application_protocol="postgresql")}, env_vars={"POSTGRES_PASSWORD": "password"}))
 --- at /home/circleci/project/core/server/api_container/server/startosis_engine/startosis_executor.go:62 (StartosisExecutor.Execute.func1) ---
Caused by: Unexpected error occurred starting service 'postgres:15.2-alpine'
 --- at /home/circleci/project/core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go:130 (AddServiceCapabilities.Execute) ---
Caused by: An error occurred while trying to get the user service container attributes for user service with UUID 'ad0ed246a194492eacf7bfea0fd0725e'
 --- at /home/circleci/project/container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/user_services_functions/start_user_services.go:320 (createStartServiceOperation.func1) ---
Caused by: An error occurred creating the user service Docker container name object
 --- at /home/circleci/project/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:215 (dockerEnclaveObjectAttributesProviderImpl.ForUserServiceContainer) ---
Caused by: An error occurred creating Docker object name from string 'postgres:15.2-alpine--ad0ed246a194492eacf7bfea0fd0725e'
 --- at /home/circleci/project/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go:447 (dockerEnclaveObjectAttributesProviderImpl.getNameForUserServiceContainer) ---
Caused by: Object name 'postgres:15.2-alpine--ad0ed246a194492eacf7bfea0fd0725e' doesn't match Docker docker object name regex '^[a-zA-Z0-9-._]+$'
 --- at /home/circleci/project/container-engine-lib/lib/backend_impls/docker/object_attributes_provider/docker_object_name/docker_object_name.go:23 (CreateNewDockerObjectName) ---

Error encountered running Starlark code.
```

### Desired behavior

Right now, the error is quite obtuse, talking about "Docker objects" and so on. Much better would be something saying that my service name doesn't pass validation.

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description:

check service name contains allowed characters and error cleanly

## Is this change user facing?
YES

## References (if applicable):
Closes #164

### CODE REVIEW

**Comment by gbouv** [comment]
If you want to run the extra mile, we also have this one: https://github.com/kurtosis-tech/kurtosis-private/issues/1167
It makes it super painful to run a script on k8s VS docker, especially wakurtosis. And since those folks are going to try k8s sometime soon, that's something we will need to address sooner rather than later.
Not needed right now strictly speaking thoug

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go [hunk 1]
```diff
@@ -101,7 +101,13 @@ func (builtin *AddServiceCapabilities) Interpret(arguments *builtin_argument.Arg
 		return nil, interpretationErr
 	}
 
-	builtin.serviceName = service.ServiceName(serviceName.GoString())
+	serviceNameGoStr := serviceName.GoString()
+
+	if isValidServiceName := service.IsServiceNameValid(serviceNameGoStr); !isValidServiceName {
+		return nil, startosis_errors.NewInterpretationError("Service name '%v' is invalid as it contains disallowed characters. Service names can only contain characters 'a-z', 'A-Z', '0-9', '-' & '_'", serviceNameGoStr)
+	}
```
Comment (gbouv) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go]: Should we push this validation into the ServiceConfig object instead, as a field validator? This way we don't have to duplicate this bit in both add_service and add_services. Wdyt?

Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go]: The ServiceConfig object doesn't hold a service_name. Am I interpreting this incorrectly?

Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go]: Have added this to the validator in `add_service_shared` as discussed on Slack!

File: internal_testsuites/golang/testsuite/startosis_add_service_test/startosis_add_service_invalid_name_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		memory_allocation=256,
+		ports = {
+			"grpc": PortSpec(number = GRPC_PORT, transport_protocol = "TCP")
+		}
+	)
+	plan.add_service(service_name = "%s", config = config)
+`
+)
+
+func TestAddServiceWithInvalidServiceNameFailsValidation(t *testing.T) {
```
Comment (h4ck3rk3y) [inline internal_testsuites/golang/testsuite/startosis_add_service_test/startosis_add_service_invalid_name_test.go:32]: added an integration test as I couldn't find a good spot to test Validation errors apart from this

Comment (gbouv) [reply internal_testsuites/golang/testsuite/startosis_add_service_test/startosis_add_service_invalid_name_test.go:32]: Yeah we don't have a good place to test validation mechanism right now... :(

**Review by gbouv — APPROVED** [review]

**Pull request merged**